packages/chip/src/Chip/Chip.styles.ts

```diff
@@ -39,175 +46,270 @@ export const chipWrapperPadding: Record<
+  [Theme.Light]: palette.gray.base,
 };
 
 /**
  * Chip wrapper
  */
-export const chipWrapperThemeStyle = (variant: Variant, theme: Theme) => css`
-  color: ${variantColor[variant][theme].color};
-  background-color: ${variantColor[variant][theme].bgColor};
-  transition: background-color ${transitionDuration.faster}ms ease-in-out;
-`;
-
-export const chipWrapperBaseStyle = css`
-  display: inline-flex;
-  align-items: center;
-  overflow: hidden;
-  white-space: nowrap;
-  box-sizing: border-box;
-  border-radius: 4px;
-`;
-
-export const chipWrapperSizeStyle = (baseFontSize: BaseFontSize) => css`
-  font-size: ${fontSize[baseFontSize]}px;
-  line-height: ${lineHeight[baseFontSize]}px;
-`;
-
-export const chipWrapperBaseDisabledStyles = css`
-  cursor: not-allowed;
-`;
-
-export const chipWrapperDisabledStyle: Record<Theme, string> = {
-  [Theme.Light]: css`
-    color: ${palette.gray.base};
-    background-color: ${palette.gray.light3};
-  `,
-  [Theme.Dark]: css`
-    color: ${palette.gray.dark2};
-    background-color: ${palette.gray.dark4};
-    box-shadow: inset 0 0 1px 1px ${palette.gray.dark2};
-  `,
-};
+export const getChipWrapperStyles = (
+  baseFontSize: BaseFontSize,
+  variant: Variant,
+  theme: Theme,
+  isDisabled = false,
+) =>
+  cx(
+    css`
+      display: inline-flex;
+      align-items: center;
+      overflow: hidden;
+      white-space: nowrap;
+      border-radius: 4px;
+      font-size: ${fontSize[baseFontSize]}px;
+      line-height: ${lineHeight[baseFontSize]}px;
+
+      color: ${variantColor[variant][theme].text.default};
+      background-color: ${variantColor[variant][theme].background.default};
+      transition: background-color ${transitionDuration.faster}ms ease-in-out;
+    `,
+    {
+      [css`
+        cursor: not-allowed;
+        background-color: ${color[theme].background.secondary.default};
+        color: ${wrapperDisabledColor[theme]};
+      `]: isDisabled,
+      [css`
+        box-shadow: inset 0 0 1px 1px ${palette.gray.dark2};
+      `]: isDisabled && theme === Theme.Dark,
+    },
+  );
 
 /**
  * Chip text
  */
-export const chipTextSizeStyle = (baseFontSize: BaseFontSize) => css`
-  padding-inline: ${chipWrapperPadding[baseFontSize].x}px;
-  padding-block: ${chipWrapperPadding[baseFontSize].y}px;
-`;
-
-export const chipTextDismissSizeStyle = css`
-  padding-inline-end: 2px;
-`;
-
-export const chipTextStyles = (variant: Variant, theme: Theme) => css`
-  &:focus-within {
-    background-color: ${variantColor[variant][theme].focusBgColor};
-  }
-
-  .${chipInlineDefinitionClassName} {
-    &:focus-visible,
-    &:focus {
-      outline: none;
-    }
-  }
-`;
-
-export const chipTextDisabledStyles: Record<Theme, string> = {
-  [Theme.Dark]: css`
-    &:focus-within {
-      background-color: ${palette.gray.dark1};
-    }
-  `,
-  [Theme.Light]: css`
-    &:focus-within {
-      background-color: ${palette.gray.light1};
-    }
-  `,
-};
+export const getChipTextStyles = (
```

Comment:
same here, it feels harder to read/maintain a big block of css vs breaking it up into human readable variables